tests/rest/client/test_sync.py

```diff
@@ -4904,6 +4884,63 @@ def test_incremental_sync_full_state_never(self) -> None:
             exact=True,
         )
 
+    def test_rooms_with_no_updates_do_not_come_down_incremental_sync(self) -> None:
+        """
+        Test that rooms with no updates are returned in subsequent incremental
+        syncs.
+        """
+
+        user1_id = self.register_user("user1", "pass")
+        user1_tok = self.login(user1_id, "pass")
+
+        room_id1 = self.helper.create_room_as(user1_id, tok=user1_tok)
+
+        sync_body = {
+            "lists": {
+                "foo-list": {
+                    "ranges": [[0, 1]],
+                    "required_state": [],
+                    "timeline_limit": 0,
+                }
+            }
+        }
+
+        _, after_room_token = self.do_sync(sync_body, tok=user1_tok)
+
+        # Make the Sliding Sync request
+        response_body, _ = self.do_sync(
+            sync_body, since=after_room_token, tok=user1_tok
+        )
```

Comment:
(align with other tests)

```suggestion
        _, from_token = self.do_sync(sync_body, tok=user1_tok)

        # Make the incremental Sliding Sync request
        response_body, _ = self.do_sync(
            sync_body, since=from_token, tok=user1_tok
        )
```